### PR TITLE
repo: always use host release and arch for Ubuntu

### PR DESCRIPTION
**Pull request by cjp256**
- Remove dependency of ProjectOptions parameter.

- Update format_sources_list() tp query release arch/release
  separately.

Update tests accordingly.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/internal/repo/_deb.py [hunk 1]
```diff
@@ -706,7 +696,10 @@ def _get_local_sources_list():
     return sources
 
 
-def _format_sources_list(sources_list: str, *, deb_arch: str, release: str = "xenial"):
+def _format_sources_list(sources_list: str):
+    release = os_release.OsRelease().version_codename()
+    deb_arch = snapcraft.ProjectOptions().deb_arch
```
Comment (sergiusens) [inline snapcraft/internal/repo/_deb.py:701]: I wonder if this can go away completely.

Comment (cjp256) [reply snapcraft/internal/repo/_deb.py]: done, i ditched $suffix and $prefix.  I was originally going to wait on that, but it's here now. :D

**Comment by sergiusens** [comment]
btw, you have format issues

**Comment by cjp256** [comment]
Fixed

**Pull request merged**